_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
+let $formattedErrorMessage := "Assert Equal failed (ASSERT-EQUAL-FAILED): a c d a b c"
+return 
+try{
+  test:assert-same-values($leftSequence, $rightSequence)
+} catch($err) {
+  test:assert-equal($err//error:format-string/string(), $formattedErrorMessage)
+}
+```
+### Scenario 5 - Test assert equal xml - all attributes and elements matched
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ $err
+}
+```
+### Scenario 6 - Test assert equal xml - attributes mis-match and test expected error
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="expected" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $errorMessageExpected := "Assert Equal failed (ASSERT-EQUAL-FAILED): actual expected  :  mismatched attribute text ($expected= /t:test/@type , $actual= /t:test/@type )"
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
+```
+#### Scenario 7 - Test assert equal xml - element value mis-match and test expected error
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+declare namespace error = "http://marklogic.com/xdmp/error";
 
-*TODO: is this information current?*
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abcd</code></t:test>
+let $errorMessageExpected := "Assert Equal failed (ASSERT-EQUAL-FAILED): abc abcd  :  mismatched element text ($expected= /t:test/code , $actual= /t:test/code )"
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
+```
+### Scenario 8 - Test assert equal JSON - same json node and values as json object
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
```

Comment:
Let's use the constructors instead of maps, it's a bit more XQuery-like and easier to read:

```
let $expected := object-node {
  "firstProperty": "first value",
  "secondProperty": "second value"
}

Reply:
The scenario description says, "same json node and values as json object". JSON Nodes and JSON objects are different structures, so I'm a bit unclear what we're going for here. If we're looking to compare JSON objects, I'd go with:

```xquery
let $actual := json:object()
    => map:with("a", 111)
    => map:with("b", 222)
```

If we're working with JSON Nodes, then it would be as @jamesagardner suggests. (Or maybe the goal is to compare a Node to an Object, in which case you'd use one for `$actual` and the other for `$expected`.